src/main/java/organice/commons/util/LevenshteinDistanceUtil.java

```diff
@@ -0,0 +1,83 @@
+package organice.commons.util;
+
+import static java.util.Objects.requireNonNull;
+import static organice.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiFunction;
+
+/**
+ * Helper functions for calculating Levenshtein Distance (edit distance). Edit distance is the minimum number of
+ * character changes required to make two strings equal.
+ */
+public class LevenshteinDistanceUtil {
+
+    /**
+     * Returns the minimum Levenshtein Distance of every {@code stringFromList} and {@code otherString} pair. If
+     * {@code stringFromList} is a single-word string, {@code otherString} is split according to spaces (if possible)
+     * and the minimum of the minimum distances of every possible pair is returned.
+     */
+    public static int findMinLevenshteinDistance(List<String> stringList, String otherString) {
+        // If keyword is one word long, split the personName and find minLD.
+        // Else just do as we are normally doing in original finalMinLD
+        BiFunction<String, String, Integer> findDistanceSplitIfMultiWord = (stringFromList, oString) ->
+                stringFromList.split(" ").length == 1 ? Arrays.stream(oString.split(" "))
+                        .reduce(Integer.MAX_VALUE, (minDistance, nextWord) -> Integer.min(minDistance,
```

Comment:
Can I confirm that the algorithm works this way for minLD:
If there are two strings 'Hans Johnson' and 'Hans Jane', it splits the two strings into two string arrays ['Hans', 'Johnson'] and ['Hans', 'Jane']. Then calculates the minLD for: ('Hans', 'Hans'), ('Hans', 'Jane') and ('Johnson', 'Hans'), ('Johnson', 'Jane')?